recipes/android-ndk/all/test_package/conanfile.py

```diff
@@ -22,10 +23,13 @@ def build(self):
             cmake.build()
 
     def test(self):
-        if self.settings.os == "Windows":
-            self.run("ndk-build.cmd --version")
-        else:
-            self.run("ndk-build --version")
+        # INFO: can_run allows mac M1, but it does not work for the NDK
+        # https://github.com/android/ndk/issues/1299
+        if can_run(self) and not (self.settings.os == "Macos" and self.settings.arch == "armv8"):
```

Comment:
Apologies you run into this on CI.

The logic here is not entirely correct - there is an issue with `ndk-build` which is specific to versions r23b and _earlier_. r23c is, for the most part, universal binaries, and have no issues running on arm64 macOS. For r23c and newer, this is not a problem.

The `can_run(self)` will skip the test if a user is running `conan create --build-require`, or `conan test` with a profile where the host profile is for Android - where in fact. We don't want this test to be skipped if possible

I propose:

```
        ndk_build = "ndk-build.cmd" if self.settings.os == "Windows" else "ndk-build"
        ndk_version = Version(self.tested_reference_str.split('/')[1])
        skip_run = platform.system() == "Darwin" and platform.processor() == "arm" and ndk_version < "r23c"
        if not skip_run:
            self.run(f"{ndk_build} --version", env="conanbuild")
        else:
            self.output.warning(f"Skipped running ndk-build on macOS Apple Silicon in arm64 mode, please use a newer"
                                 " version of the Android NDK")
```

where

* it skips the test if we detect we are on macOS ARM and the version is older r23b or older - for newer versions we run the tests. Note that I'm querying python, and not Conan - this is correct here, Conan _could_ be theortically running on an interpreter in x86_64 mode, and the test would then pass.
* add `env=conanbuild`, which is correct since the test is consuming the ndk as a `tool-requires`

Reply:
Your explanation makes sense to me, but the CI changes were actually made by @uilianries, so maybe they should review them too?

Separately does a similar change need to be made to `test_v1_package`?